### PR TITLE
PLT-7: Refactoring frontend (chunk 8)

### PR DESCRIPTION
**Pull request by enahum**
- Sidebar and related components
- Small Tweak to demotion and add msg for terminal cmd

### CODE REVIEW

File: web/react/components/unread_channel_indicator.jsx [hunk 1]
```diff
@@ -31,5 +31,5 @@ UnreadChannelIndicator.defaultProps = {
 UnreadChannelIndicator.propTypes = {
     show: React.PropTypes.bool,
     extraClass: React.PropTypes.string,
-    text: React.PropTypes.string
+    text: React.PropTypes.object
```
Comment (hmhealey) [inline web/react/components/unread_channel_indicator.jsx]: This is fine as it is, but for future reference, if you're passing a React element as a prop, you can use React.PropTypes.node to be a bit more specific than using a plain object.

**Comment by hmhealey** [comment]
+1

File: web/react/components/unread_channel_indicator.jsx (hunk 1 above)
Comment (enahum) [inline web/react/components/unread_channel_indicator.jsx]: Ok great. thx @hmhealey 

I could change it, that's not a problem really.

**Comment by jwilander** [comment]
+1

**Pull request merged**